### PR TITLE
PR: Remove serialized length limit when sending modified variables back to the kernel

### LINKED ISSUE #10955: The object modified it too big to be sent back to the kernel
### What steps will reproduce the problem?

I have a nested dictionary with a size of 3033. Each key of the dictionary contains 9 items, including strings and dictionaries. I can easily modify them in the variable explorer in Spyder 3. However, When I modify them in the variable explorer in Spyder 4, it shows me a warning, "The object you are trying to modify is too big to be sent back to the kernel. Therefore, your modifications won't take place."  

## Versions

* Spyder version: 4.0.0 
* Python version: 3.7.3
* Qt version: 5.12.5
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.5.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.2 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.1 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  None (NOK)
cython >=0.21                :  None (NOK)
IPython >=4.0                :  7.10.1 (OK)
matplotlib >=2.0.0           :  3.1.2 (OK)
pandas >=0.13.1              :  0.25.3 (OK)
numpy >=1.7                  :  1.17.3 (OK)
scipy >=0.17.0               :  1.3.1 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
rtree >=0.8.3                :  0.9.1 (OK)
```

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

 - asizeof doesn't seem to work correctly:
```
In [1]: import numpy as np

In [2]: from pympler.asizeof import asizeof

In [3]: val = np.zeros(1000*1000*20)

In [4]: asizeof(val)
Out[4]: 160000096

In [5]: # modify val in the variable explorer

In [6]: asizeof(val)
Out[6]: 96

In [7]: np.shape(val)
Out[7]: (20000000,)

In [8]: 
```

 - If a variable was loaded from the kernel, it should be sent back to the kernel.
 - I am not sure spyder-ide/spyder#7158 still applies with the current architecture anyway.
 - It there must be a limit, it should be implemented at the comms level, not here.

### Issue(s) Resolved

Fixes #10955

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/variableexplorer/widgets/collectionseditor.py [hunk 1]
```diff
@@ -1438,23 +1437,12 @@ def __init__(self, parent, data, minmax=False, shellwidget=None,
     def get_value(self, name):
         """Get the value of a variable"""
         value = self.shellwidget.get_value(name)
-        # Reset temporal variable where value is saved to
-        # save memory
-        self.shellwidget._kernel_value = None
```
Comment (impact27) [inline spyder/plugins/variableexplorer/widgets/collectionseditor.py]: I think this is left over from a previous implementation as I couldn't find `_kernel_value`

**Comment by ccordoba12** [comment]
Thanks for this @impact27! About your comments:

> If a variable was loaded from the kernel, it should be sent back to the kernel.

Totally right.

> I am not sure #7158 still applies with the current architecture anyway.

Right again.

> It there must be a limit, it should be implemented at the comms level, not here.

Let's not do that for now, only if/when it's necessary.

**Comment by ccordoba12** [comment]
We'd like to include your contribution in our next release in the Spyder 4 series. For that, please make your branch to derive from our `4.x` branch (instead of master), with the following command:

    git rebase --onto 4.x master fix_variable_explorer

then push again.

File: spyder/plugins/variableexplorer/widgets/collectionseditor.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/plugins/variableexplorer/widgets/collectionseditor.py]: Yes, it was.

**Pull request merged**